drivers/gpio/gpio_shell.c

```diff
@@ -21,167 +21,161 @@
 
 LOG_MODULE_REGISTER(gpio_shell);
 
-struct args_index {
-	uint8_t port;
-	uint8_t index;
-	uint8_t mode;
-	uint8_t value;
+struct sh_gpio {
+	const struct device *dev;
+	gpio_pin_t pin;
 };
 
-static const struct args_index args_indx = {
-	.port = 1,
-	.index = 2,
-	.mode = 3,
-	.value = 3,
-};
+static void get_sh_gpio(const struct shell *sh, char **argv, int8_t arg_i,
+			struct sh_gpio *gpio)
+{
+	gpio->dev = device_get_binding(argv[arg_i]);
+	gpio->pin = atoi(argv[arg_i + 1]);
+}
 
-static int cmd_gpio_conf(const struct shell *sh, size_t argc, char **argv)
+static int conf_cmd_handler(const struct shell *sh,
+			    size_t argc, char **argv, void *data)
 {
-	uint8_t index = 0U;
-	int type = GPIO_OUTPUT;
-	const struct device *dev;
+	struct sh_gpio gpio;
+	gpio_flags_t flags;
+	int ret;
 
-	if (isdigit((unsigned char)argv[args_indx.index][0]) != 0 &&
-	    isalpha((unsigned char)argv[args_indx.mode][0]) != 0) {
-		index = (uint8_t)atoi(argv[args_indx.index]);
-		if (!strcmp(argv[args_indx.mode], "in")) {
-			type = GPIO_INPUT;
-		} else if (!strcmp(argv[args_indx.mode], "inu")) {
-			type = GPIO_INPUT | GPIO_PULL_UP;
-		} else if (!strcmp(argv[args_indx.mode], "ind")) {
-			type = GPIO_INPUT | GPIO_PULL_DOWN;
-		} else if (!strcmp(argv[args_indx.mode], "out")) {
-			type = GPIO_OUTPUT;
-		} else {
-			return 0;
-		}
-	} else {
-		shell_error(sh, "Wrong parameters for conf");
-		return -ENOTSUP;
-	}
+	get_sh_gpio(sh, argv, -3, &gpio);
 
-	dev = device_get_binding(argv[args_indx.port]);
+	flags = (intptr_t)data;
 
-	if (dev != NULL) {
-		index = (uint8_t)atoi(argv[args_indx.index]);
-		shell_print(sh, "Configuring %s pin %d",
-			    argv[args_indx.port], index);
-		gpio_pin_configure(dev, index, type);
+	ret = gpio_pin_configure(gpio.dev, gpio.pin, flags);
+	if (ret != 0) {
+		shell_error(sh, "error: %d", ret);
+		return ret;
 	}
 
 	return 0;
 }
 
-static int cmd_gpio_get(const struct shell *sh,
-			size_t argc, char **argv)
+SHELL_SUBCMD_DICT_SET_CREATE(sub_conf, conf_cmd_handler,
+	(in, GPIO_INPUT, "input"),
+	(inu, (GPIO_INPUT | GPIO_PULL_UP), "input with pull up"),
+	(ind, (GPIO_INPUT | GPIO_PULL_DOWN), "input with pull down"),
```

Comment:
Non-blocking suggestion:
```suggestion
	(in_pu, (GPIO_INPUT | GPIO_PULL_UP), "input with pull up"),
	(in_pd, (GPIO_INPUT | GPIO_PULL_DOWN), "input with pull down"),
```